### PR TITLE
DM-6992: Use Bokeh views entirely for dashboard

### PR DESCRIPTION
**Pull request by afausti**
DM-6992 extends SQUASH dashboard to handle multiple datasets it also changes the API to decouple bokeh from django models. Now it is possible retrieve the jobs filtering by dataset like:

`http://localhost:8000/dashboard/api/jobs/?ci_dataset=cfht`

and there is another endpoint to retrieve the datasets:

`http://localhost:8000/dashboard/api/datasets/`

I did a major rewrite of the metrics app and implemented the dataset and metric selection using bokeh widgets, it also plots the specification thresholds for each metric.

Deployment depends on DM-7098 to include the new attributes `ci_dataset` and `ci_label` in the JSON created by post_qa .

### CODE REVIEW

File: squash/dashboard/models.py [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
     """Job information"""
     STATUS_OK = 0
     STATUS_FAILED = 1
 
-    ci_name = models.CharField(max_length=32, blank=False,
-                               help_text='Name of the Jenkins job')
     ci_id = models.CharField(max_length=16, blank=False,
                              help_text='Jenkins job ID')
+    ci_name = models.CharField(max_length=32, blank=False,
+                               help_text='Name of the Jenkins project,\
```
Comment (jonathansick) [inline squash/dashboard/models.py]: An elegant way of dealing with multi-line strings that are wrapped in parentheses:

```
CharField(help_text='Name of the Jenkins project, '
                                   'e.g. validate_drp')
```

Python auto-concatenates those two strings.

**Comment by jonathansick** [comment]
Some UI suggestions:

napkin 16-08-11 11 17 52 am

**Comment by afausti** [comment]
Thanks @jonathansick for your suggestions. I agree that a drop down will do a better work there. As for the labels I've used the default style in bokeh 0.12 - but I can change that and also increase the font size (as we have currently in production). I liked your suggestion of having a "next" button, let me see if I can implement that as part of this PR, otherwise will open another ticket for that.

**Comment by afausti** [comment]
Improved layout, also tried to use span annotations 

http://bokeh.pydata.org/en/0.12.1/docs/user_guide/annotations.html#spans

for the metric thresholds but it cannot be configured by a datasource and so does not respond correctly to the widget changes, will figure out a solution later and will merge these changes for now.

**Pull request merged**